aws/resource_aws_storagegateway_nfs_file_share.go

```diff
@@ -89,26 +97,28 @@ func resourceAwsStorageGatewayNfsFileShare() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"directory_mode": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "0777",
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "0777",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-7]{4}$`), ""),
```

Comment:
Maybe refactor this to a shared function in `validators.go`?

Reply:
good point, will do.

Reply:
Done.